pkg/util/winutil/iisconfig/apmtags.go

```diff
@@ -0,0 +1,103 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+//go:build windows
+
+// Package iisconfig manages iis configuration
+package iisconfig
+
+/*
+the file datadog.json can be located anywhere; it is path-relative to a .net application
+give the path name, read the json and return it as a map of string/string
+*/
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"os"
+)
+
+type APMTags struct {
+	DDService string
+	DDEnv     string
+	DDVersion string
+}
+
+// ReadDatadogJSON reads a datadog.json file and returns the APM tags
+func ReadDatadogJSON(datadogJSONPath string) (APMTags, error) {
+	var datadogJSON map[string]string
+	var apmtags APMTags
+
+	file, err := os.Open(datadogJSONPath)
+	if err != nil {
+		return apmtags, err
+	}
+	defer file.Close()
+
+	decoder := json.NewDecoder(file)
+	err = decoder.Decode(&datadogJSON)
+	if err != nil {
+		return apmtags, err
+	}
+	apmtags.DDService = datadogJSON["DD_SERVICE"]
+	apmtags.DDEnv = datadogJSON["DD_ENV"]
+	apmtags.DDVersion = datadogJSON["DD_VERSION"]
+	return apmtags, nil
+}
+
+type iisAppSetting struct {
+	Key   string `xml:"key,attr"`
+	Value string `xml:"value,attr"`
+}
+type iisAppSettings struct {
+	XMLName xml.Name        `xml:"appSettings"`
+	Adds    []iisAppSetting `xml:"add"`
```

Comment:
Curious, can this be a map if search is expected?
If there is no search on this, a list is fine.

Reply:
I'm not following.  it's a list because that's what the xml is providing.  the `xml` tag allows the xml decoder to marshal the xml straight into the structure, so the structure needs to follow the xml.